beater/request/context.go

```diff
@@ -54,7 +56,7 @@ func (c *Context) Reset(w http.ResponseWriter, r *http.Request) {
 	c.Logger = nil
 	c.TokenSet = false
 	c.Authorized = false
-
+	// TODO i guess i should not do reset the rate limit manager ??
```

Comment:
It needs to be reset. The middleware ensures the rate limiter is set for every request. The context is reused for any other request. What if it is reused for a request to a different endpoint with a different rate limiter?

Reply:
ok. i don't get it, thou. the rate limiter is stateful, if i reset it every request then is not the case that nothing will ever be rate limited?

Reply:
You do create the rate limiter instance once https://github.com/elastic/apm-server/pull/2706/files#diff-7b23454a544752fbe51da51555207091R30. 
For every request that is wrapped in the same middleware the same rate limiter is then set https://github.com/elastic/apm-server/pull/2706/files#diff-7b23454a544752fbe51da51555207091R34.

Reply:
right, thanks